Create basic alertbuffer client

zgalor opened the pull request:
basic client with get functionality

zgalor commented:
@moolitayer @yaacov @ilackarms
This is still WIP but I would appreciate your feedback

zgalor commented:
@moolitayer @yaacov @ilackarms bump

ilackarms commented on lib/prometheus/alert_client/client.rb:
```diff
@@ -0,0 +1,116 @@
+        return unless headers && headers[:token]
+
+        {
+          Authorization: 'Bearer ' + headers[:token].to_s,
```
nit: `"Bearer #{headers[:token].to_s}"`

moolitayer commented on examples/get_alerts.rb:
```diff
@@ -0,0 +1,10 @@
+require 'prometheus/alert_client'
+
+# returns a client
+prometheus = Prometheus::AlertClient.client(url: 'http://localhost:9099', path: '/topics/zgalor')
```
I prefer the kubeclient style, it is more idiomatic:
```
client = Kubeclient::Client.new(...)
```
so
```
Prometheus::AlertClient.new(..)
```

Also, and unlike the kubeclient, I think it's great that you use keyword arguments!
Definitely one of the things we have learned from experience would make future changes easier

zgalor replied:
It's the same. You can access it the same way:
`Prometheus::AlertClient::Client.new(...)`

I could cancel the layering and have `PrometheusAlertClient::Client.new(...)`
but this way it's more consistent with `Prometheus::ApiClient`

moolitayer replied:
@zgalor I suggested 
```
Prometheus::AlertClient.new(..)
```

Up to you...

moolitayer commented on examples/get_alerts.rb:
```diff
@@ -0,0 +1,10 @@
+# returns a client
+prometheus = Prometheus::AlertClient.client(url: 'http://localhost:9099', path: '/topics/zgalor')
+
+alerts = prometheus.get('12497ca8-b597-4590-ac5d-d55af7f3d185', 2)
```
Since kw arguments are great for sensibility, consider making all the exposed methods use them

moolitayer commented on lib/prometheus/alert_client/client.rb:
```diff
@@ -0,0 +1,116 @@
+# encoding: UTF-8
+
+require 'json'
+require 'faraday'
```
Really up to you but take a look at the discussion in
https://github.com/abonas/kubeclient/issues/237
and consider
https://github.com/nahi/httpclient 

faraday should be okay though, and is extensively used in ManageIQ.

moolitayer commented on lib/prometheus/alert_client/client.rb:
```diff
@@ -0,0 +1,116 @@
+
+      # Default paramters for creating default client
+      DEFAULT_ARGS = {
+        url: 'http://localhost:9099',
```
url should probably not have a default

moolitayer commented:
@zgalor might be worth separating Faraday related stuff to a separate file, up to you

zgalor commented:
@moolitayer @ilackarms @yaacov 
I added basic spec, gemfile, and README.
Also renamed to alert_buffer_client.
Removed WIP.
This is ready on my side.
If you LGTM this I will squash, merge and create the initial Gem

ilackarms commented:
lgtm!

zgalor commented:
squashed and fixed some newline issues.
Merging
